### ISSUE TITLE
containerize 

### ISSUE THREAD

**Issue by dmd** [mercure-imaging/mercure#6]
I think it would be a *really* good idea to containerize; it would take setup from "90% likely to go wrong" to "cannot go wrong".

If I get some time, I'll work on it - hard to justify the time for myself now that I have Hermes working just fine. 

Some of what would have to happen:

* create a `Dockerfile` that does the basic install
* create a `docker-compose.yml` which creates, I think, 6 containers - one for each of the services currently being managed by systemd - all mounting the same volume
  *  I'm not sure the best way to deal with multiple dispatch workers
* some minor modifications to remove the UI around starting/stopping services, conditional on whether we detect we're running under Docker
* will have to figure out how to best deal with secrets generation

**Comment by joshy** [comment]
Definitely agree that a containerized version would be easier to install and use. I am not sure if there needs to be 6 different containers. Why not start with one container which contains everything?  
* It needs to expose one port for receiving dicoms
* Multiple folders needs to be mounted into docker, `incoming`,`outgoing` etc (or maybe just the parent folder `hermes-data`) for processing
* It needs to expose one port for the web app
* One folder for the configuration

Open / unclear
 * How many dispatch workers are needed needs to be changed on the fly
 * secrets generation

**Comment by dmd** [comment]
I've generally been under the impression that [there should be one top-level running process per container](https://devops.stackexchange.com/questions/447/why-it-is-recommended-to-run-only-one-process-in-a-container)

**Comment by RobertFischer** [comment]
Nope: one _function_ per container.  You can have as many top-level processes as you want, although you have to watch out for the PID 1 problem.  https://github.com/phusion/baseimage-docker#docker_single_process
